### PR TITLE
Code cleanup around Pollers

### PR DESCRIPTION
**Pull request by Spikhalskiy**
Fix for potential pollers deadlock if suspendPolling called twice.
Refactorings that make the code around Polling easier to follow.

### CODE REVIEW

File: temporal-sdk/src/main/java/io/temporal/internal/worker/ActivityPollTask.java [hunk 1]
```diff
@@ -110,6 +110,7 @@ public ActivityTask poll() {
               ProtobufTimeUtils.toM3Duration(
                   response.getStartedTime(), response.getCurrentAttemptScheduledTime()));
       isSuccessful = true;
+      return new ActivityTask(response, pollSemaphore::release);
```
Comment (Spikhalskiy) [inline temporal-sdk/src/main/java/io/temporal/internal/worker/ActivityPollTask.java:113]: Paired with 
```
if (!isSuccessful) pollSemaphore.release();
```
in finally, this location of `return` is cleaner for the reader.

File: temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java [hunk 2]
```diff
@@ -168,15 +168,18 @@ public void awaitTermination(long timeout, TimeUnit unit) {
 
   @Override
   public void suspendPolling() {
-    log.info("suspendPolling: {}", this);
-    suspendLatch.set(new CountDownLatch(1));
```
Comment (Spikhalskiy) [inline temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java:172]: Absense of CAS here may replace and lose the latch which is already used by Pollers to block on. This way they will never be unblocked if suspentPolling was called twice.

File: temporal-sdk/src/main/java/io/temporal/internal/worker/SingleWorkerOptions.java [hunk 3]
```diff
@@ -126,35 +126,33 @@ public Builder setDefaultHeartbeatThrottleInterval(Duration defaultHeartbeatThro
     }
 
     public SingleWorkerOptions build() {
+      PollerOptions pollerOptions = this.pollerOptions;
       if (pollerOptions == null) {
-        pollerOptions =
-            PollerOptions.newBuilder()
-                .setPollBackoffInitialInterval(Duration.ofMillis(200))
```
Comment (Spikhalskiy) [inline temporal-sdk/src/main/java/io/temporal/internal/worker/SingleWorkerOptions.java:132]: This code path is dead. We come here only in some tests where we do
```
        SingleWorkerOptions.newBuilder().build()
```

and these options and not important for these tests. As a result, these magic values 200 and 20 here are never used and beter be removed from the production code paths.

File: temporal-sdk/src/main/java/io/temporal/worker/Worker.java [hunk 4]
```diff
@@ -181,7 +181,7 @@ private static SingleWorkerOptions toLocalActivityOptions(
       List<ContextPropagator> contextPropagators,
       Scope metricsScope) {
     return toSingleWorkerOptions(factoryOptions, options, clientOptions, contextPropagators)
-        .setPollerOptions(PollerOptions.newBuilder().build())
```
Comment (Spikhalskiy) [inline temporal-sdk/src/main/java/io/temporal/worker/Worker.java:184]: ActivityWorker and WorkflowWorker PollerOptions specify this parameter explicitly. 
To simplify the untangling for the reader, let's specify it for LocalActivityWorker explicitly too. 1 is the default value for pollThreadCount used inside PollerOptions.

**Review by Sushisource — APPROVED** [review]

**Pull request merged**